src/exercises/concurrency/chat-app.md

```diff
@@ -0,0 +1,123 @@
+# Broadcast Chat Application
+
+In this exercise, we want to use our new knowledge to implement a broadcast
+chat application. We have a chat server that the clients connect to and publish
+their messages. The client reads user messages from the standard input, and
+sends them to the server. The chat server broadcasts each message that it
+receives to all the clients.
+
+For this, we use [a broadcast channel][1] on the server, and
+[`tokio_websockets`][2] for the communication between the client and the
+server.
+
+Create a new Cargo project and add the following dependencies:
+
+`Cargo.toml`:
+
+<!-- File Cargo.toml -->
+
+```toml
+[package]
+name = "chat-async"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+futures-util = "0.3.28"
+http = "0.2.9"
+thiserror = "1.0.40"
+tokio = { version = "1.28.1", features = ["net", "macros", "time", "rt", "rt-multi-thread", "io-std", "io-util"] }
```

Comment:
I haven't checked the feature list, but I remember that there is a [`full` feature](https://tokio.rs/tokio/tutorial/hello-tokio#the-code) which Tokio uses in their own tutorial. If it covers at least these features, I would suggest using it for simplicity (I believe we care more about simplicity than compile times).

Reply:
Yes. "full" works. Done.